src/pages/net-neutrality/petition.js

```diff
@@ -9,26 +9,44 @@ import SignupForm from '../../components/net-neutrality/signup-form.js';
 import FormBody from '../../components/signup-form/form-body.js';
 import Modal from '../../components/modal.js';
 
+const DONATE_CTA_DELAY = 1000; // in milliseconds
+
 var Signup = React.createClass({
   getInitialState: function() {
     return {
       showModal: false,
+      dismissedModal: false,
       signupSuccess: false
     };
   },
   componentDidMount: function() {
-    /*if (!this.props.location.query.subscribed) {
-      setTimeout(() => {
-        this.setState({
-          showModal: true
-        });
-      }, 3000);
-    }*/
+    if (typeof window !== "undefined" && window.document) {
+      this._withScroll = e => this.handleScroll(e);
+      window.addEventListener('scroll', this._withScroll);
+    }
+  },
+  componentWillUnmount() {
+    if (this._withScroll) {
+      window.removeEventListener('scroll', this._withScroll);
+    }
+  },
+  handleScroll: function(e) {
+    if (this._ctaTimeout) {
+      clearTimeout(this._ctaTimeout);
+    }
+    this._ctaTimeout = setTimeout(() => this.spawnDonateCTA(), DONATE_CTA_DELAY);
+  },
+  spawnDonateCTA: function() {
+    if (!sessionStorage.getItem('dismissedModal')) {
```

Comment:
probably want to check if `sessionStorage` is accessible or not (in case they've blocked cookies)